File: touyou/GitMe/GitMe/Domain/Login/LoginConverter.swift
//
//  LoginConverter.swift
//  GitMe
//
//  Created by 藤井陽介 on 2018/01/22.
//  Copyright © 2018 touyou. All rights reserved.
//

import UIKit
import RxSwift

protocol LoginConverterProtocol {

    func logIn() -> Observable<UserInfoViewModel>
}

class LoginConverter {

    // MARK: Private

    private let api = GitHubAPI.shared
}

// MARK: - Protocol Interface

extension LoginConverter: LoginConverterProtocol {

    func logIn() -> Observable<UserInfoViewModel> {

        return api.logIn().map { user in

            let userInfoViewModel = UserInfoViewModel()
            userInfoViewModel.userName = user.login
            userInfoViewModel.iconUrl = user.avatarUrl
            return userInfoViewModel
        }
    }
}

File: touyou/GitMeAndroid/app/src/main/java/com/dev/touyou/gitme/Domain/Login/LoginConverter.kt
package com.dev.touyou.gitme.Domain.Login

import com.dev.touyou.gitme.Data.GitHubAPI
import com.dev.touyou.gitme.Domain.UserInfoViewModel
import io.reactivex.Observable

/**
 * Created by touyou on 2018/02/26.
 */

interface LoginConverterInterface {

    fun logIn(): Observable<UserInfoViewModel>
}

class LoginConverter: LoginConverterInterface {

    override fun logIn(): Observable<UserInfoViewModel> {

        return GitHubAPI.logIn().map {

            val userInfoViewModel = UserInfoViewModel()
            userInfoViewModel.userName = it.login
            userInfoViewModel.iconUrl = it.avatarUrl
            userInfoViewModel
        }
    }
}